src/Adapter/Customer/Group/Repository/GroupRepository.php

```diff
@@ -82,6 +85,38 @@ public function assertGroupExists(GroupId $groupId): void
      */
     public function create(CustomerGroup $customerGroup): GroupId
     {
-        return new GroupId($this->addObjectModelToShops($customerGroup, $customerGroup->id_shop_list, CannotAddGroupException::class));
+        $groupId = $this->addObjectModelToShops(
+            $customerGroup,
+            array_map(fn (int $shopId) => new ShopId($shopId), $customerGroup->id_shop_list),
+            CannotAddGroupException::class
+        );
+
+        return new GroupId($groupId);
+    }
+
+    /**
+     * @param int $customerGroupId
+     *
+     * @return int[]
+     */
+    public function getAssociatedShopIds(int $customerGroupId): array
+    {
+        return $this->getObjectModelAssociatedShopIds($customerGroupId, CustomerGroup::class);
+    }
+
+    /**
+     * @param CustomerGroup $customerGroup
+     */
```

Comment:
This one is unecessary too but you need to add a phpdoc for the array
```suggestion
    /**
     * @param string[] $propertiesToUpdate
     */
```

Reply:
You're right I'll update this in the next PR